crates/hyperqueue/src/worker/start/program.rs

```diff
@@ -210,18 +210,18 @@ async fn resend_stdio(
     job_task_id: JobTaskId,
     channel: ChannelId,
     stdio: Option<impl tokio::io::AsyncRead + Unpin>,
-    stream: Rc<StreamSender>,
+    stream: StreamSender,
 ) -> tako::Result<()> {
     if let Some(mut stdio) = stdio {
-        log::debug!("Starting stream {}/{}/{}", job_id, job_task_id, channel);
+        log::debug!("Resending stream {}/{}/{}", job_id, job_task_id, channel);
         loop {
             let mut buffer = vec![0; STDIO_BUFFER_SIZE];
             let size = stdio.read(&mut buffer[..]).await?;
+            buffer.truncate(size);
```

Comment:
Buffer with size zero is used to signal the end of the stream? It would be nice to leave an explicit comment about this. Or even an explicit enum variant in `StreamerMessage` :)